tests/test_reader.py

```diff
@@ -43,3 +54,4 @@ def test_example_data():
 
         assert isinstance(read_data, Template)
         assert validate_data_dict(template, read_data, root)
+        Writer(read_data, nxdl_file, tmp_output).write()
```

Comment:
Is there any way to test the written file? E.g., have a test .nxs file and check that the contents are the same? I want to also try that for the XPS reader.

Reply:
This is a good idea to check the required fields and attributes in `.nxs` file. Which ultimately verifies the required contents.
I will extend the test.

Reply:
Here I added another part in test to check the content of nexus file.